migration/idempotent/008-tracing-functions.sql

```diff
@@ -386,6 +392,7 @@ AS $func$
         SELECT a.key_id, a.id
         FROM _ps_trace.tag a
         WHERE x.key = a.key
+        AND pg_catalog.sha512(x.value::TEXT::BYTEA) = pg_catalog.sha512(a.value::TEXT::BYTEA)
         AND x.value = a.value
```

Comment:
I think we should remove `AND x.value = a.value` so we can avoid the heap lookup

Reply:
This and other redundant comparisons are safeguards against hash collisions. AFAIK heap lookups would happen for index hits only, so I'd rather stay on the "correct but slightly slower" side.